tests/mutation_test.cc

```diff
@@ -1860,3 +1860,36 @@ SEASTAR_THREAD_TEST_CASE(test_collection_compaction) {
     BOOST_CHECK(!cmut.tomb);
     BOOST_CHECK(cmut.cells.empty());
 }
+
+// Reproducer for #4567: "appending_hash<row> ignores cells after first null"
+SEASTAR_THREAD_TEST_CASE(test_appending_hash_row_4567) {
```

Comment:
We could also extend `generate_mutation_sets()` to generate some null columns. We already have unit tests for hashes which are using `for_each_mutation()`.